models/image.py

```diff
@@ -905,44 +714,70 @@ def from_ref_and_new(cls, ref, new):
 
         Parameters
         ----------
-        ref: Image object
-            The reference image to use.
-        new: Image object
+        new_image: Image object
             The new image to use.
-
+        ref_image: Image object
+            The reference image to use.
+        im_type: str (optional)
+            If given, will also set the output image's type.
+            Use "diff" or "warped", which will be prepended
+            with "Com" if the new image is a "Com..." type image
+            Note that if not given, the output type will remain as "Sci".
         Returns
         -------
         output: Image
             The new Image object. It would not have any data variables or filepath.
         """
+        if ref_image is None:
+            raise ValueError("Must provide a reference image.")
+        if new_image is None:
+            raise ValueError("Must provide a new image.")
+
         output = Image(nofile=True)
 
         # for each attribute, check the two images have the same value
-        for att in ['section_id', 'instrument', 'telescope', 'filter', 'project', 'target']:
-            ref_value = getattr(ref, att)
-            new_value = getattr(new, att)
+        for att in ['instrument', 'telescope', 'project', 'section_id', 'filter', 'target']:
+            ref_value = getattr(ref_image, att)
+            new_value = getattr(new_image, att)
 
             if att == 'section_id':
                 ref_value = str(ref_value)
                 new_value = str(new_value)
-            if ref_value != new_value:
+
+            if att in ['section_id', 'filter', 'target'] and ref_value != new_value:
                 raise ValueError(
                     f"Cannot combine images with different {att} values: "
                     f"{ref_value} and {new_value}. "
                 )
-            output.__setattr__(att, new_value)
+
+            # assign the values from the new image
+            setattr(output, att, new_value)
         # TODO: should RA and Dec also be exactly the same??
 
+        output.upstream_images = [ref_image, new_image]
+        output.ref_image = ref_image
+        output.ref_image_id = ref_image.id
```

Comment:
Doesn't setting output.ref_image also set the ref_image.id?

But also, note my comments above where I suggest removing the ref_image and new_image fields from Image, and putting the nature of the assocation in the image assocation table.